org.eclipse.jdt.core.manipulation/proposals/org/eclipse/jdt/internal/ui/text/correction/SuppressWarningsSubProcessorCore.java

```diff
@@ -38,4 +72,204 @@ public static final boolean hasSuppressWarningsProposal(IJavaProject javaProject
 		return false;
 	}
 
+	public static Collection<ASTRewriteCorrectionProposalCore> addSuppressWarningsProposals(IInvocationContextCore context, IProblemLocationCore problem) {
+		Collection<ASTRewriteCorrectionProposalCore> proposals = new ArrayList<>();
+		if (problem.isError() && !JavaCore.ENABLED.equals(context.getCompilationUnit().getJavaProject().getOption(JavaCore.COMPILER_PB_SUPPRESS_OPTIONAL_ERRORS, true))) {
+			return null;
+		}
+		if (JavaCore.DISABLED.equals(context.getCompilationUnit().getJavaProject().getOption(JavaCore.COMPILER_PB_SUPPRESS_WARNINGS, true))) {
+			return null;
+		}
+
+		String warningToken= CorrectionEngine.getWarningToken(problem.getProblemId());
+		if (warningToken == null) {
+			return null;
+		}
+		for (ChangeCorrectionProposalCore element : proposals) {
+			if (element instanceof SuppressWarningsProposalCore && warningToken.equals(((SuppressWarningsProposalCore) element).getWarningToken())) {
+				return null; // only one at a time
+			}
+		}
+
+		ASTNode node= problem.getCoveringNode(context.getASTRoot());
+		if (node == null) {
+			return null;
+		}
+
+		ASTNode target= node;
+		int relevance= IProposalRelevance.ADD_SUPPRESSWARNINGS;
+		do {
+			relevance= addSuppressWarningsProposalIfPossible(context.getCompilationUnit(), target, warningToken, relevance, proposals);
+			if (relevance == 0)
+				return proposals;
+			target= target.getParent();
+		} while (target != null);
+
+		ASTNode importStatement= ASTNodes.getParent(node, ImportDeclaration.class);
+		if (importStatement != null && !context.getASTRoot().types().isEmpty()) {
+			target= (ASTNode) context.getASTRoot().types().get(0);
+			if (target != null) {
+				addSuppressWarningsProposalIfPossible(context.getCompilationUnit(), target, warningToken, IProposalRelevance.ADD_SUPPRESSWARNINGS, proposals);
+			}
+		}
+		return proposals;
+	}
+
+	/**
+	 * Adds a SuppressWarnings proposal if possible and returns whether parent nodes should be
+	 * processed or not (and with what relevance).
+	 *
+	 * @param cu the compilation unit
+	 * @param node the node on which to add a SuppressWarning token
+	 * @param warningToken the warning token to add
+	 * @param relevance the proposal's relevance
+	 * @param proposals collector to which the proposal should be added
+	 * @return <code>0</code> if no further proposals should be added to parent nodes, or the
+	 *         relevance of the next proposal
+	 *
+	 * @since 3.6
```

Comment:
FYI - since tags on internal classes and methods are not required. Furthermore as it's copied from another bundle with different version it's plain wrong.

Reply:
Thanks for letting me know. Removed it.